### PR TITLE
[topic-gpio] boards: stm32: update to use new GPIO API

### PR DESCRIPTION
**Pull request by erwango**
Update all STM32 based boards to use new GPIO API.
Based on: #19579 ([topic-gpio] drivers: gpio_stm32: update to use new GPIO API).

Due to number of impacted boards, Shippable will likely timeout.
So Shippable result should be checked in #19579. And only usual code review is possible in this PR.

### CODE REVIEW

File: boards/arm/nucleo_f103rb/nucleo_f103rb.dts [hunk 1]
```diff
@@ -30,10 +30,37 @@
 		compatible = "gpio-keys";
 		user_button: button {
 			label = "User";
-			gpios = <&gpioc 13 GPIO_INT_ACTIVE_LOW>;
+			gpios = <&gpioc 13 GPIO_ACTIVE_LOW>;
 		};
```
Comment (avisconti) [inline boards/arm/nucleo_f103rb/nucleo_f103rb.dts]: Do not get why these two changes are in this commit and not in 717de74

File: boards/arm/disco_l475_iot1/disco_l475_iot1.dts [hunk 2]
```diff
@@ -37,7 +37,7 @@
 		compatible = "gpio-keys";
 		user_button: button_0 {
 			label = "User";
-			gpios = <&gpioc 13 GPIO_INT_ACTIVE_LOW>;
+			gpios = <&gpioc 13 GPIO_ACTIVE_LOW>;
 		};
 	};
```
Comment (avisconti) [inline boards/arm/disco_l475_iot1/disco_l475_iot1.dts]: not in 717de74 ?

File: drivers/gpio/gpio_stm32.c [hunk 3]
```diff
@@ -41,24 +45,37 @@ static void gpio_stm32_isr(int line, void *arg)
  */
 const int gpio_stm32_flags_to_conf(int flags, int *pincfg)
 {
-	int direction = flags & GPIO_DIR_MASK;
-	int pud = flags & GPIO_PUD_MASK;
```
Comment (avisconti) [inline drivers/gpio/gpio_stm32.c]: This is a code cleaning, right?
Shouldn't be put in a separate commit rather than keep it in "new GPIO API" commit?

File: drivers/gpio/gpio_stm32.c [hunk 4]
```diff
@@ -41,24 +45,37 @@ static void gpio_stm32_isr(int line, void *arg)
...
+
+		if ((flags & GPIO_PULL_UP) != 0) {
 			*pincfg |= STM32_PINCFG_PULL_UP;
-		} else if (pud == GPIO_PUD_PULL_DOWN) {
+		} else if ((flags & GPIO_PULL_DOWN) != 0) {
 			*pincfg |= STM32_PINCFG_PULL_DOWN;
 		} else {
-			/* floating */
 			*pincfg |= STM32_PINCFG_FLOATING;
 		}
```
Comment (avisconti) [inline drivers/gpio/gpio_stm32.c]: code cleaning?

File: drivers/gpio/gpio_stm32.c [hunk 5]
```diff
@@ -201,7 +218,7 @@ static inline uint32_t gpio_stm32_pin_to_exti_line(int pin)
 
 static void gpio_stm32_set_exti_source(int port, int pin)
 {
-	uint32_t line = gpio_stm32_pin_to_exti_line(pin);
+	u32_t line = gpio_stm32_pin_to_exti_line(pin);
```
Comment (avisconti) [inline drivers/gpio/gpio_stm32.c]: code cleaning

File: drivers/gpio/gpio_stm32.c [hunk 6]
```diff
@@ -227,7 +244,7 @@ static void gpio_stm32_set_exti_source(int port, int pin)
 
 static int gpio_stm32_get_exti_source(int pin)
 {
-	uint32_t line = gpio_stm32_pin_to_exti_line(pin);
+	u32_t line = gpio_stm32_pin_to_exti_line(pin);
```
Comment (avisconti) [inline drivers/gpio/gpio_stm32.c]: code cleaning again?

**Review by avisconti — CHANGES_REQUESTED** [review]
I'll finish the review later on.
I will also test it on my boards and samples

File: boards/arm/nucleo_f103rb/nucleo_f103rb.dts (hunk 1 above)
Comment (erwango) [reply boards/arm/nucleo_f103rb/nucleo_f103rb.dts]: Indeed, I pushed these initially for testing purpose.
I'll likely reshuffle the commits once all tests are done.

File: drivers/gpio/gpio_stm32.c (hunk 3 above)
Comment (erwango) [reply drivers/gpio/gpio_stm32.c]: No this is linked to the fact I'm using `flags` below and don't need `direction` and `pud` anymore.
So directly linked to new API.

File: drivers/gpio/gpio_stm32.c (hunk 4 above)
Comment (erwango) [reply drivers/gpio/gpio_stm32.c]: ok

File: drivers/gpio/gpio_stm32.c (hunk 5 above)
Comment (erwango) [reply drivers/gpio/gpio_stm32.c]: Indeed, not directly linked, but reported by CI.

File: drivers/gpio/gpio_stm32.c (hunk 6 above)
Comment (erwango) [reply drivers/gpio/gpio_stm32.c]: Indeed, not directly linked, but reported by CI.

File: boards/arm/nucleo_f103rb/nucleo_f103rb.dts [hunk 7]
```diff
@@ -21,7 +21,7 @@
 	leds {
 		compatible = "gpio-leds";
 		green_led_2: led_2 {
-			gpios = <&gpioa 5 GPIO_INT_ACTIVE_HIGH>;
+			gpios = <&gpioa 5 0>;
```
Comment (ydamigos) [inline boards/arm/nucleo_f103rb/nucleo_f103rb.dts]: GPIO_ACTIVE_HIGH?

Comment (pabigot) [reply boards/arm/nucleo_f103rb/nucleo_f103rb.dts]: Active high is the default; previous discussion reached (undocumented?) consensus that only non-default flags would be specified in devicetree.

We should really confirm and document that.  @galak @mnkp is that in fact the standard position?

File: boards/arm/nucleo_g071rb/nucleo_g071rb.dts [hunk 8]
```diff
@@ -22,7 +22,7 @@
 	leds {
 		compatible = "gpio-leds";
 		green_led_1: led_4 {
-			gpios = <&gpioa 5 GPIO_INT_ACTIVE_LOW>;
+			gpios = <&gpioa 5 GPIO_ACTIVE_LOW>;
```
Comment (ydamigos) [inline boards/arm/nucleo_g071rb/nucleo_g071rb.dts]: Is the LED actually active low?

File: boards/arm/nucleo_f103rb/nucleo_f103rb.dts (hunk 7 above)
Comment (galak) [reply boards/arm/nucleo_f103rb/nucleo_f103rb.dts]: I'd personally prefer that we show the flags in dts as its more clear if you are reading it to know.

Comment (ydamigos) [reply boards/arm/nucleo_f103rb/nucleo_f103rb.dts]: Also in linux DT they show the flags in dts (e.g. [eds-gpio.txt](https://www.kernel.org/doc/Documentation/devicetree/bindings/leds/leds-gpio.txt))

Comment (pabigot) [reply boards/arm/nucleo_f103rb/nucleo_f103rb.dts]: OK; there are a lot in linux that are `0` but there are also a lot that are `1` so `0` may not be indicating "default".

I too think explicit is better, but could live with zero for default.  Either way we should try to be consistent.  Hopefully my recollection that somebody had strong preference for leaving it zero for default was a false memory.

Comment (erwango) [reply boards/arm/nucleo_f103rb/nucleo_f103rb.dts]: Will set `GPIO_ACTIVE_HIGH`. I prefer to be explicit as well.

File: boards/arm/nucleo_g071rb/nucleo_g071rb.dts (hunk 8 above)
Comment (erwango) [reply boards/arm/nucleo_g071rb/nucleo_g071rb.dts]: Sounds strange, will x-check.

**Comment by erwango** [comment]
Squashed into https://github.com/zephyrproject-rtos/zephyr/pull/19579.
Closing
